Node/src/TeamsChatConnector.ts

```diff
@@ -155,7 +159,85 @@ export class TeamsChatConnector extends builder.ChatConnector {
         } else {
-          callback(new Error('Failed to authorize request'), null);
+          return callback(new Error('Failed to authorize request'), null);
+        }
+    });
+  }
+
+  /**
+  *  Return a newly started reply chain address in channel
+  *  @param {string} serverUrl - Server url is composed of baseUrl and cloud name, remember to find your correct cloud name in session or the function will not find the team.
+  *  @param {string} channelId - The channel id, will post in the channel.  
+  *  @param {builder.IMessage|builder.IIsMessage} message - The message to post in the channel.
+  *  @param {function} callback - This callback returns err or result.
+  */
+  public startReplyChain(serverUrl: string, channelId: string, message: builder.IMessage|builder.IIsMessage, callback?: (err: Error, address: builder.IChatConnectorAddress) => void) : void {
+    var options: msRest.RequestOptions = {customHeaders: {}, jar: false};
+    var restClient = new RestClient(serverUrl, null);
+    var remoteQuery = new RemoteQuery(restClient);
+    this.getAccessToken((err, token) => {
+        if (!err && token) {
+          options.customHeaders = {
+            'Authorization': 'Bearer ' + token
+          };
+
+          var innerCallback = function (err: Error, result: ReplyResult) {
+
+            if (!callback)
+            {
+              return;
+            }
+
+            if (result && result.hasOwnProperty("id") && result.hasOwnProperty("activityId"))
+            {
+              var messageAddress = null;
+              if ((<builder.IIsMessage>message).toMessage) {
+                messageAddress = (<builder.IIsMessage>message).toMessage().address;
+              }
+              else if ((<builder.IMessage>message).address)
+              {
+                messageAddress = (<builder.IMessage>message).address;
+              }
+              else
+              {
+                throw new Error("Message type is wrong. Need either Message or IMessage or IIsMessage");
+              }
+
+              var address: builder.IChatConnectorAddress = <builder.IChatConnectorAddress>{
+                conversation: {},
+                bot: {}
+              };
+              address.channelId = 'msteams';
+              address.conversation.id = result.id;
+              address.id = result.activityId;
+              address.bot.id = messageAddress.bot.id;
+              address.bot.name = messageAddress.bot.name;
+
+              if (address.user) {
+                  delete address.user;
+              }
+
+              return callback(null, address);          
+            }
+            else
+            {
+              let error = new Error("Failed to start reply chain: no conversation ID and activity ID returned.");
+              return callback(error, null);            
+            }
+          } 
+
+          if ((<builder.IIsMessage>message).toMessage) {
```

Comment:
This check happens twice, I recommend refactoring so we convert so IMessage in the beginning and work with that.